extensions/terminal-suggest/src/terminalSuggestMain.ts

```diff
@@ -112,10 +112,16 @@ export async function resolveCwdFromPrefix(prefix: string, currentCwd?: vscode.U
 		return;
 	}
 	try {
-		// Resolve the absolute path of the prefix
-		const resolvedPath = path.resolve(currentCwd?.fsPath, prefix);
+		// Get the nearest folder path from the prefix. This ignores everything after the `/` as
+		// they are what triggers changes in the directory.
+		// TODO: Support `\` on relevant shells
+		const lastSlashIndex = prefix.lastIndexOf('/');
```

Comment:
you can use pathSeparatorhttps://github.com/microsoft/vscode/blob/168e235635b3975031eb073d974947f42633d713/extensions/terminal-suggest/src/terminalSuggestMain.ts#L96

Reply:
It's more nuanced that that as shell type comes into play too, but this is an easy fix to improve it :+1: